src/Workspaces/Core/Portable/CodeFixes/FixAllOccurrences/FixAllContext.DiagnosticProvider.cs

```diff
@@ -31,6 +35,30 @@ public abstract class DiagnosticProvider
             /// This includes both document-level diagnostics for all documents in the given project and project-level diagnostics, i.e. diagnostics with no source location, in the given project. 
             /// </summary>
             public abstract Task<IEnumerable<Diagnostic>> GetAllDiagnosticsAsync(Project project, CancellationToken cancellationToken);
+
+            internal virtual Task<ImmutableDictionary<Document, ImmutableArray<Diagnostic>>> GetDocumentDiagnosticsToFixAsync(
```

Comment:
If we remove the methods then FixAllContext.DiagnosticProvider.GetDocumentDiagnosticsToFixAsync won't have anything it can call on FixAllProvider.

FixAllContext.DaignosticProvider.GetDocumentDiagnosticsToFixAsync is called FixAllContext.GetDocumentDiagnosticsToFixAsync  and that is called from the supression code.  And that only has a FixAllProvider to work with.

Note: i am very very very much against random downcasting to derived types to get different behavior and to bypass the normal class behavior.  That's an abuse of OO and should not be done in that matter.

The current approach (while not attractive) demonstrates where the actual difference in behavior is actually manifested, and does not require other types to know and downcast the class hierarchy of a different part of the system.